File: ESTRUTURAS/3160/main.dart
import 'dart:io';

String newList(List<String> l, List<String> n, String x) {
  if (l.contains(x)) {
    int i = l.indexOf(x);
    l.insertAll(i, n);
    return l.join(' ');
  }
  return (l + n).join(' ');
}

void main() {
  List<String> l = stdin.readLineSync()!.trim().split(' ');
  List<String> n = stdin.readLineSync()!.trim().split(' ');
  String x = stdin.readLineSync()!.trim();
  print(newList(l, n, x));
}

File: ESTRUTURAS/3160/Main.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;

public class Main {

    static String newList(ArrayList<String> l, ArrayList<String> n, String x) {
        if (l.contains(x)) {
            int i = l.indexOf(x);
            l.addAll(i, n);
            return String.join(" ", l);
        }
        l.addAll(n);
        return String.join(" ", l);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        ArrayList<String> l = new ArrayList<String>(Arrays.asList(scanner.nextLine().trim().split(" ")));
        ArrayList<String> n = new ArrayList<String>(Arrays.asList(scanner.nextLine().trim().split(" ")));
        String x = scanner.next();
        System.out.println(newList(l, n, x));
        scanner.close();
    }
}